cli/crates/server/src/servers.rs

```diff
@@ -13,31 +13,74 @@ use common::consts::MAX_PORT;
 use common::consts::{GRAFBASE_DIRECTORY_NAME, GRAFBASE_SCHEMA_FILE_NAME, GRAFBASE_TS_CONFIG_FILE_NAME};
 use common::environment::{Environment, Project, SchemaLocation};
 use common::types::UdfKind;
+use engine::registry::Registry;
 use flate2::read::GzDecoder;
 use futures_util::FutureExt;
 use std::borrow::Cow;
 use std::collections::HashMap;
 use std::env;
-use std::net::Ipv4Addr;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 use std::path::Path;
 use std::sync::atomic::Ordering;
-use std::sync::mpsc::{self, Receiver, Sender};
 use std::sync::Arc;
 use std::time::{Duration, SystemTime};
-use std::{
-    fs,
-    process::Stdio,
-    thread::{self, JoinHandle},
-};
+use std::{fs, process::Stdio};
 use tokio::net::TcpListener;
 use tokio::process::Command;
-use tokio::runtime::Builder;
 use tokio::sync::broadcast::{self, channel};
 use version_compare::Version;
 use which::which;
 
 const EVENT_BUS_BOUND: usize = 5;
 
+pub async fn production_start(
+    listen: IpAddr,
+    port: u16,
+    tracing: bool,
+    message_sender: tokio::sync::mpsc::UnboundedSender<ServerMessage>,
+) -> Result<(), ServerError> {
+    use bridge::BridgeState;
+
+    export_embedded_files()?;
+    create_project_dot_grafbase_directory()?;
+    validate_dependencies().await?;
```

Comment:
Do we still need this if you don't use UDFs?

Reply:
we use UDFs, It's not great as I'm creating an instance of miniflare per resolver right now, but we do use it. Have started a PR to have a single miniflare, but didn't finish.

Reply:
I mean, could we do this only if UDFs are in use?

Reply:
but yeah fair point

Reply:
yeah definitely, didn't properly check what it was doing, good catch :)

Reply:
Do we need that for the SDK though?
We might also be able to do the export given that condition

Reply:
ah agreed again. Keeping `create_project_dot_grafbase_directory()?` though, as it's not necessarily tied to resolvers, no?

Reply:
Yeah that's used for the DB as well. I'd just check that we also include a check for the SDK in the condition since you'll need node for it and the extract likely

Reply:
what do you mean with SDK? the sdk inside resolvers will only be used if there are resolvers. So we're good with the UDF check, no?

Reply:
I mean if you use the TS config, that uses node and IIRC an asset

Reply:
ahhh you're right, doesn't use assets but does use node obviously.

Reply:
But I'm assuming we can check for that

Reply:
Well moved the check behind udf check and doing it in the schema parsing if we're using node. Double-check, but it's simple. 🤷